In [2]:
# Importing the libraries 
import sys
import numpy as np 
import pandas as pd 
import os 

### Table: TABLE Frame with the questions we have

In [3]:
#Sub table 1 
data1 = {'Question':['Modes of communicating with target high-risk populations (elderly, health care workers).', 
                    'Management of patients who are underhoused or otherwise lower socioeconomic status.', 
                    'What are ways to create hospital infrastructure to prevent nosocomial outbreaks and protect uninfected patients?', 
                    'Methods to control the spread in communities, barriers to compliance',
                    'What are recommendations for combating/overcoming resource failures',
                    'Other relevant sources']}
tbl1 = pd.DataFrame(data1)


#Sub table 2
data2 = {'From Abstacts file':['Q1_articles_Ways_of_communicating_with_highrisk_pop.csv', 'Q2_Mangmnt_underhoused_or_low_income_patients.csv', 'Q3_Articles_Ways_for_hospitals_prevent_nosocomial_outbreaks_and_protect_uninfected.csv', 'Q4_Articles_Methods_control_spread_in_communities_and_barriers_to_compliance.csv','Q5_Journal_Recommendations_for_overcoming_COVID19_resource_failures.csv','COVID19_Journal_Abrstracts.csv'], 
        'Online sources file':['Q1_online_Ways_of_communicating_with_highrisk_pop.csv', 'Q2_online_Managing_homeless_or_lowIncome_patients.csv','Q3_online_Ways_hospital_infrastr_prevent_nosocomial_outbreaks_and_protect_uninfected.csv', 'Q4_online_Methods_control_community_spread_and_barriers_to_compliance.csv','Q5_online_Recommendations_for_overcoming_COVID19_resource_failures.csv', 'COVID19_Population_Studies_online.csv']}
tbl2 = pd.DataFrame(data2)

In [4]:
#Join the two tables together 
tablesTable=tbl1.join(tbl2)
tablesTable.head(10)

Question  \
0  Modes of communicating with target high-risk p...   
1  Management of patients who are underhoused or ...   
2  What are ways to create hospital infrastructur...   
3  Methods to control the spread in communities, ...   
4  What are recommendations for combating/overcom...   
5                             Other relevant sources   

                                  From Abstacts file  \
0  Q1_articles_Ways_of_communicating_with_highris...   
1  Q2_Mangmnt_underhoused_or_low_income_patients.csv   
2  Q3_Articles_Ways_for_hospitals_prevent_nosocom...   
3  Q4_Articles_Methods_control_spread_in_communit...   
4  Q5_Journal_Recommendations_for_overcoming_COVI...   
5                     COVID19_Journal_Abrstracts.csv   

                                 Online sources file  
0  Q1_online_Ways_of_communicating_with_highrisk_...  
1  Q2_online_Managing_homeless_or_lowIncome_patie...  
2  Q3_online_Ways_hospital_infrastr_prevent_nosoc...  
3  Q4_online_Methods_control_community_spread_and...  
4  Q5_online_Recommendations_for_overcoming_COVID...  
5              COVID19_Population_Studies_online.csv

### Table: ABSTRACTS

In [6]:
#Import the information
df=pd.read_csv('E:/NorthEastern University/XN-Hackathon Covid/Kaggle Data/metadata.csv') 
print('The number of articles are:',df.shape[0])

C:\Users\HP\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3063: DtypeWarning: Columns (5,6,13,14,16) have mixed types.Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


The number of articles are: 167470


In [5]:
#Select only title, abstract, date, link to the article, and authors
tablesAbstract= df [['title', 'abstract', 'publish_time', 'url', 'authors']]

In [6]:
#Seperate the words in the Abstract columns 
tablesAbstract['words']=  tablesAbstract.abstract.str.strip().str.split('[\W_]+')
tablesAbstract = tablesAbstract[tablesAbstract.words.str.len() > 0]
                                        #rid of all the input that have the Words column less than 0 
tablesAbstract.head(3)

C:\Users\HP\Anaconda3\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


title  \
0  Clinical features of culture-proven Mycoplasma...   
1  Nitric oxide: a pro-inflammatory mediator in l...   
2    Surfactant protein-D and pulmonary host defense   

                                            abstract publish_time  \
0  OBJECTIVE: This retrospective chart review des...   2001-07-04   
1  Inflammatory diseases of the respiratory tract...   2000-08-15   
2  Surfactant protein-D (SP-D) participates in th...   2000-08-25   

                                                 url  \
0  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...   
1  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...   
2  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...   

                                             authors  \
0                Madani, Tariq A; Al-Ghamdi, Aisha A   
1  Vliet, Albert van der; Eiserich, Jason P; Cros...   
2                                    Crouch, Erika C   

                                               words  
0  [OBJECTIVE, This, retrospective, chart, review...  
1  [Inflammatory, diseases, of, the, respiratory,...  
2  [Surfactant, protein, D, SP, D, participates, ...

In [7]:
print('The number of articles after reduction is:', tablesAbstract.shape[0])

The number of articles after reduction is: 122492


# ==> We will analyze what the articles available in the dataset and then also search online resources for each questions 

# -------------------------------------------------------------------------------------------------------------

# 0/ General information about the COVID 

Scraping the web using beautifulsoup4 and Google 

In [8]:
pip install google

Note: you may need to restart the kernel to use updated packages.


In [9]:
pip install beautifulsoup4

Note: you may need to restart the kernel to use updated packages.


In [10]:
try: 
    from googlesearch import search 
except ImportError:  
    print("Error/Not found") 
                        ### browsing for: COVID 19 population studies  
query = "COVID 19 population studies"
                        ##limit to the first 10 relevant links:
for j in search(query, tld="co.in", num=10, stop=10, pause=2): 
    print(j)  

https://www.cdc.gov/coronavirus/2019-ncov/covid-data/serology-surveillance/index.html
https://www.cdc.gov/coronavirus/2019-ncov/cases-updates/geographic-seroprevalence-surveys.html
https://www.dhs.wisconsin.gov/news/releases/061720.htm
https://www.washington.edu/populationhealth/covid-19/
https://www.washington.edu/populationhealth/covid-19/covid-19-economic-recovery-research-grants/
https://www.washington.edu/populationhealth/covid-19/covid-19-livestream-talks/
https://healthitanalytics.com/news/population-health-study-to-estimate-prevalence-of-covid-19-in-ca
https://www.thelancet.com/journals/lancet/article/PIIS0140-6736(20)31483-5/fulltext
https://www.thelancet.com/journals/langlo/article/PIIS2214-109X(20)30264-3/fulltext
https://medium.com/microbial-instincts/population-studies-confirm-risk-factors-for-catching-covid-19-853779e7a3c6


In [11]:
#Gather all the links we found online to one dataframe 
COVID_G_Studies={'url': ['https://www.cdc.gov/coronavirus/2019-ncov/covid-data/serology-surveillance/index.html',
    'https://www.cdc.gov/coronavirus/2019-ncov/cases-updates/geographic-seroprevalence-surveys.html',
    'https://www.dhs.wisconsin.gov/news/releases/061720.htm',
    'https://www.washington.edu/populationhealth/covid-19/',
    'https://www.washington.edu/populationhealth/covid-19/covid-19-economic-recovery-research-grants/',
    'https://www.washington.edu/populationhealth/covid-19/covid-19-livestream-talks/',
    'https://healthitanalytics.com/news/population-health-study-to-estimate-prevalence-of-covid-19-in-ca',
    'https://www.thelancet.com/journals/lancet/article/PIIS0140-6736(20)31483-5/fulltext',
    'https://www.thelancet.com/journals/langlo/article/PIIS2214-109X(20)30264-3/fulltext',
    'https://medium.com/microbial-instincts/population-studies-confirm-risk-factors-for-catching-covid-19-853779e7a3c6']}

COVID_G_Studies=pd.DataFrame.from_dict(COVID_G_Studies)
COVID_G_Studies

url
0  https://www.cdc.gov/coronavirus/2019-ncov/covi...
1  https://www.cdc.gov/coronavirus/2019-ncov/case...
2  https://www.dhs.wisconsin.gov/news/releases/06...
3  https://www.washington.edu/populationhealth/co...
4  https://www.washington.edu/populationhealth/co...
5  https://www.washington.edu/populationhealth/co...
6  https://healthitanalytics.com/news/population-...
7  https://www.thelancet.com/journals/lancet/arti...
8  https://www.thelancet.com/journals/langlo/arti...
9  https://medium.com/microbial-instincts/populat...

In [12]:
#Save the dataframe to the below 
COVID_G_Studies.to_csv('E:/NorthEastern University/XN-Hackathon Covid/Kaggle Data/Q0_Gen_COVID19_Population_Studies.csv') 

# 1/ Modes of communicating with target high-risk populations (elderly, health care workers).

In [13]:
#Create different sets with specialized terms  from our questions 
Q1_a = tablesAbstract[tablesAbstract['abstract'].str.contains('communicating')]
Q1_b = tablesAbstract[tablesAbstract['abstract'].str.contains('with high-risk')]
Q1_c = tablesAbstract[tablesAbstract['abstract'].str.contains('elderly')]
Q1_d = tablesAbstract[tablesAbstract['abstract'].str.contains('elders')]
                                        #choose elders and elderly because if we choose only one then the number of articles is lower 
Q1_e = tablesAbstract[tablesAbstract['abstract'].str.contains('communicating')]
Q1_f = tablesAbstract[tablesAbstract['abstract'].str.contains('health care workers')]
Q1_g = tablesAbstract[tablesAbstract['abstract'].str.contains('risk population')]


In [14]:
#Concatenitng the dataframes into one table per question and dropping null value 
Q1 = pd.concat([Q1_a,Q1_b,Q1_c,Q1_d,Q1_e,Q1_f,Q1_g])
Q1.dropna(inplace=True)
Q1.shape

(2462, 6)

We have 2462 articles regarding to the this topic at the moment

In [15]:
#Articles that we have
Q1.head(5)

title  \
472   A molecular clamp ensures allosteric coordinat...   
739   Making sense of perceptions of risk of disease...   
2947  A mechanistic spatio-temporal framework for mo...   
3107  Health-seeking behavior and transmission dynam...   
4132  Epidemiological and clinical profile of Korean...   

                                               abstract publish_time  \
472   Although the ribosome is mainly comprised of r...   2010-07-21   
739   BACKGROUND: Maintaining high levels of childho...   2011-12-20   
2947  In recent years there has been growing availab...   2017-10-30   
3107  BACKGROUND: It has been found that health-seek...   2018-03-06   
4132  The aim of this study was to investigate the e...   2019-09-27   

                                                    url  \
472   https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2...   
739   https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...   
2947  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...   
3107  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...   
4132  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6...   

                                                authors  \
472            Meskauskas, Arturas; Dinman, Jonathan D.   
739                          Bond, Lyndal; Nolan, Terry   
2947  Lau, Max S. Y.; Gibson, Gavin J.; Adrakey, Hol...   
3107     You, Shu-Han; Chen, Szu-Chieh; Liao, Chung-Min   
4132          Kim, Jiena; Choi, Hyo Jeong; Kim, Ho Jung   

                                                  words  
472   [Although, the, ribosome, is, mainly, comprise...  
739   [BACKGROUND, Maintaining, high, levels, of, ch...  
2947  [In, recent, years, there, has, been, growing,...  
3107  [BACKGROUND, It, has, been, found, that, healt...  
4132  [The, aim, of, this, study, was, to, investiga...

In [16]:
## Saved as CSV file
Q1.to_csv('E:/NorthEastern University/XN-Hackathon Covid/Kaggle Data/Q1_Ways_of_communicating_with_highrisk_pop.csv')

Online Search 

In [44]:
try: 
    from googlesearch import search 
except ImportError:  
    print("Error/Not found") 
### Browsing the sourses using different search terms: 
search1_1 = "ways to reach COVID 19 vulnerable population"
search1_2 = "communicating with COVID 19 high risk population"
search1_3 = "telehealth and COVID19"

##limit to the first 10 relevant links:
for a in search(search1_1, tld="co.in", num=10, stop=10, pause=2): 
    print(a)
for a in search(search1_2, tld="co.in", num=10, stop=10, pause=2): 
    print(a)
for a in search(search1_3, tld="co.in", num=10, stop=10, pause=2): 
    print(a)

https://www.cdc.gov/coronavirus/2019-ncov/need-extra-precautions/index.html
https://www.cdc.gov/coronavirus/2019-ncov/need-extra-precautions/caregivers-dementia.html
https://www.cdc.gov/coronavirus/2019-ncov/need-extra-precautions/racial-ethnic-minorities.html
https://www.cdc.gov/coronavirus/2019-ncov/need-extra-precautions/people-with-medical-conditions.html
https://www.cdc.gov/healthyweight/assessing/bmi/index.html
https://www.cdc.gov/diabetes/managing/flu-sick-days.html
https://www.cdc.gov/coronavirus/2019-ncov/community/retirement/checklist.html
https://www.cdc.gov/asthma/triggers.html
https://sph.washington.edu/news-events/news/partnering-state-health-department-reach-vulnerable-communities-about-covid-19
https://www.ncoa.org/blog/managing-the-covid-19-crisis-for-vulnerable-populations/
https://www.cdc.gov/coronavirus/2019-ncov/communication/index.html
https://www.cdc.gov/coronavirus/2019-ncov/communication/print-resources.html
https://www.cdc.gov/coronavirus/2019-ncov/communicati

In [41]:
#Manually choose the current relevant sources 
Q1_Online_Source={'url': ['https://www.cdc.gov/coronavirus/2019-ncov/need-extra-precautions/index.html',
                         'https://www.cdc.gov/coronavirus/2019-ncov/need-extra-precautions/caregivers-dementia.html',
                         'https://www.cdc.gov/coronavirus/2019-ncov/need-extra-precautions/racial-ethnic-minorities.html',
                         'https://www.cdc.gov/coronavirus/2019-ncov/need-extra-precautions/people-with-medical-conditions.html',
                         'https://www.cdc.gov/coronavirus/2019-ncov/community/retirement/checklist.html',
                         'https://www.cdc.gov/asthma/triggers.html',
                         'https://www.cdc.gov/coronavirus/2019-ncov/communication/print-resources.html?Sort=Date%3A%3Adesc',
                         'https://www.cdc.gov/coronavirus/2019-ncov/communication/videos.html?Sort=Date%3A%3Adesc',
                         'https://www.cdc.gov/coronavirus/2019-ncov/need-extra-precautions/communication-toolkit.html',
                         'https://www.hhs.gov/coronavirus/telehealth/index.html',
                         'https://sph.washington.edu/news-events/news/partnering-state-health-department-reach-vulnerable-communities-about-covid-19',
                         'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7158804/',
                         'https://www.cchpca.org/resources/covid-19-telehealth-coverage-policies',
                        'https://www.nejm.org/doi/full/10.1056/nejmp2003539',
                        'https://www.acponline.org/practice-resources/covid-19-practice-management-resources/telehealth-coding-and-billing-during-covid-19'
]}

In [53]:
Q1_Online_Source=pd.DataFrame.from_dict(Q1_Online_Source)
Q1_Online_Source.to_csv('E:/NorthEastern University/XN-Hackathon Covid/Kaggle Data/Q1_ZOnline_Sources.csv') 

# 2/ Management of patients who are underhoused or otherwise lower socioeconomic status.

In [17]:
#Create different sets with specialized terms  from our questions 
Q2_a = tablesAbstract[tablesAbstract['abstract'].str.contains('homeless')]
Q2_b = tablesAbstract[tablesAbstract['abstract'].str.contains('low income')]
Q2_c = tablesAbstract[tablesAbstract['abstract'].str.contains('poor')]
Q2_d = tablesAbstract[tablesAbstract['abstract'].str.contains('poverty')]
                                        #choose poor, lowincome, poverty because if we choose only one then the number of articles is lower 
Q2_e = tablesAbstract[tablesAbstract['abstract'].str.contains('housing')]

In [18]:
#Concatenitng the dataframes into one table per question and dropping null value 
Q2 = pd.concat([Q2_a,Q2_b,Q2_c,Q2_d,Q2_e])
Q2.dropna(inplace=True)
Q2.shape

(4728, 6)

We have 4728 articles regarding to the this topic at the moment

In [19]:
#Articles that we have
Q2.head(5)

title  \
298   Initial psychological responses to Influenza A...   
1938  Upstream Disaster Management to Support People...   
3015  Cage size, movement in and out of housing duri...   
5534  How did the introduction of managed care for t...   
7664  Geographic Distribution of Disaster-Specific E...   

                                               abstract publish_time  \
298   BACKGROUND: The outbreak of the pandemic flu, ...   2009-10-06   
1938  The unique context of day-to-day living for pe...   2015-08-18   
3015  Upper respiratory infection (URI) is not an in...   2018-01-02   
5534  Concerns about access under managed care have ...         2003   
7664  OBJECTIVE: We aimed to characterize the geogra...   2016-02-09   

                                                    url  \
298   https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2...   
1938  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4...   
3015  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...   
5534  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...   
7664  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...   

                                                authors  \
298   Goodwin, Robin; Haque, Shamsul; Neto, Felix; M...   
1938  Sundareswaran, Madura; Ghazzawi, Andrea; O'Sul...   
3015  Wagner, Denae C.; Kass, Philip H.; Hurley, Kat...   
5534  Ettner, Susan L.; Argeriou, Milton; McCarty, D...   
7664  Lee, David C.; Smith, Silas W.; Carr, Brendan ...   

                                                  words  
298   [BACKGROUND, The, outbreak, of, the, pandemic,...  
1938  [The, unique, context, of, day, to, day, livin...  
3015  [Upper, respiratory, infection, URI, is, not, ...  
5534  [Concerns, about, access, under, managed, care...  
7664  [OBJECTIVE, We, aimed, to, characterize, the, ...

In [20]:
## Saved as CSV file
Q2.to_csv('E:/NorthEastern University/XN-Hackathon Covid/Kaggle Data/Q2_Mangament_underhoused_or_low_income_patients.csv')

Online Search

In [45]:
try: 
    from googlesearch import search 
except ImportError:  
    print("Error/Not found") 
### Browsing the sourses using different search terms: 
search2_1 = "homeless COVID 19 patients"
search2_2 = "low income COVID 19 patients"


##limit to the first 10 relevant links:
for a in search(search2_1, tld="co.in", num=10, stop=10, pause=2): 
    print(a)
for a in search(search2_2, tld="co.in", num=10, stop=10, pause=2): 
    print(a)

https://www.cdc.gov/coronavirus/2019-ncov/community/homeless-shelters/index.html
https://www.cdc.gov/coronavirus/2019-ncov/community/homeless-shelters/unsheltered-homelessness.html
https://www.cdc.gov/coronavirus/2019-ncov/community/homeless-shelters/faqs.html
https://www.cdc.gov/coronavirus/2019-ncov/community/homeless-shelters/plan-prepare-respond.html
https://www.cdc.gov/mmwr/volumes/69/wr/mm6917e2.htm
https://kingcounty.gov/covid/homeless
https://www.npr.org/2020/07/11/890000877/arizona-non-profit-provides-housing-the-homeless-who-test-positive-for-coronavir
https://www.statnews.com/2020/04/11/coronavirus-san-francisco-homeless-doctors-difficult-choices/
https://neighborcare.org/services/homeless-and-housing/
https://www.asam.org/Quality-Science/covid-19-coronavirus/homeless-guidance
https://www.kff.org/coronavirus-covid-19/issue-brief/low-income-and-communities-of-color-at-higher-risk-of-serious-illness-if-infected-with-coronavirus/
http://www.seattle.gov/mayor/covid-19
https://ww

In [46]:
#Manually choose the current relevant sources 
Q2_Online_Source={'url': ['https://www.cdc.gov/coronavirus/2019-ncov/community/homeless-shelters/index.html',
                          'https://www.cdc.gov/coronavirus/2019-ncov/community/homeless-shelters/unsheltered-homelessness.html',
                          'https://www.cdc.gov/coronavirus/2019-ncov/community/homeless-shelters/faqs.html',
                          'https://www.cdc.gov/coronavirus/2019-ncov/community/homeless-shelters/plan-prepare-respond.html',
                          'https://www.cdc.gov/mmwr/volumes/69/wr/mm6917e2.htm',
                          'https://kingcounty.gov/depts/health/locations/homeless-health/healthcare-for-the-homeless/covid.aspx',
                          'https://www.npr.org/sections/coronavirus-live-updates/2020/07/11/890000877/arizona-non-profit-provides-housing-the-homeless-who-test-positive-for-coronavir',
                          'https://www.statnews.com/2020/04/11/coronavirus-san-francisco-homeless-doctors-difficult-choices/',
                          'https://neighborcare.org/services/homeless-and-housing/',
                          'https://www.kff.org/coronavirus-covid-19/issue-brief/low-income-and-communities-of-color-at-higher-risk-of-serious-illness-if-infected-with-coronavirus/',
                          'https://www.asam.org/Quality-Science/covid-19-coronavirus/homeless-guidance',
                          'http://www.seattle.gov/mayor/covid-19',
                          'https://www.seattle.gov/mayor/covid-19/covid-19-testing',
                          'http://www.seattle.gov/mayor/covid-19/covid-19-reopening',
                          'https://www.seattle.gov/rentinginseattle',
                          'https://nlihc.org/coronavirus-and-housing-homelessness',
                          'https://nlihc.org/covid-19-working-groupcalls',
                          'https://nlihc.org/resource/congressional-leaders-agree-coronavirus-response-package-funding-homelessness-and-housing',
                          'https://www.sciencedaily.com/releases/2020/05/200526173830.htm'
                         ]}

In [52]:
Q2_Online_Source=pd.DataFrame.from_dict(Q2_Online_Source)
Q2_Online_Source.to_csv('E:/NorthEastern University/XN-Hackathon Covid/Kaggle Data/Q2_ZOnline_Sources.csv') 

# 3/ What are ways to create hospital infrastructure to prevent nosocomial outbreaks and protect uninfected patients?

In [21]:
#Create different sets with specialized terms  from our questions 
Q3_a = tablesAbstract[tablesAbstract['abstract'].str.contains('nosocomial')]
Q3_b = tablesAbstract[tablesAbstract['abstract'].str.contains('hospital spread')]
Q3_c = tablesAbstract[tablesAbstract['abstract'].str.contains('hospital patients')]
Q3_d = tablesAbstract[tablesAbstract['abstract'].str.contains('nosocomial outbreak')]
Q3_e = tablesAbstract[tablesAbstract['abstract'].str.contains('hospital infrastructure')]

In [22]:
#Concatenitng the dataframes into one table per question and dropping null value 
Q3 = pd.concat([Q3_a,Q3_b,Q3_c,Q3_d, Q3_e])
Q3.dropna(inplace=True)
Q3.shape

(877, 6)

We have 877 articles regarding to the first topics at the moment

In [23]:
#Articles that we have
Q3.head(5)

title  \
332   Multiple lung abscesses due to acinetobacter i...   
653   Communicable Diseases Prioritized for Surveill...   
823   Quantifying Type-Specific Reproduction Numbers...   
1079  Should lower respiratory tract secretions from...   
1624  Identification of Antigenic Proteins of the No...   

                                               abstract publish_time  \
332   Acinetobacter species are well-known causes of...   2009-12-18   
653   INTRODUCTION: To establish strategic prioritie...   2011-10-04   
823   An important determinant of a pathogen's succe...   2012-04-12   
1079  INTRODUCTION: Influenza is easily overlooked i...   2012-06-14   
1624  The continuous expansion of nosocomial infecti...   2014-10-21   

                                                    url  \
332   https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2...   
653   https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...   
823   https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...   
1079  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...   
1624  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4...   

                                                authors  \
332   Kokkonouzis, Ioannis; Christou, Ioannis; Athan...   
653   Balabanova, Yanina; Gilsdorf, Andreas; Buda, S...   
823   Cooper, Ben S.; Kypraios, Theodore; Batra, Rah...   
1079  Giannella, Maddalena; Rodríguez-Sánchez, Belen...   
1624  Hoppe, Sebastian; Bier, Frank F.; von Nickisch...   

                                                  words  
332   [Acinetobacter, species, are, well, known, cau...  
653   [INTRODUCTION, To, establish, strategic, prior...  
823   [An, important, determinant, of, a, pathogen, ...  
1079  [INTRODUCTION, Influenza, is, easily, overlook...  
1624  [The, continuous, expansion, of, nosocomial, i...

In [24]:
## Saved as CSV file
Q3.to_csv('E:/NorthEastern University/XN-Hackathon Covid/Kaggle Data/Q3_Ways_for_hospitals_prevent_nosocomial_outbreaks_and_protect_uninfected.csv')

Online Search

In [49]:
try: 
    from googlesearch import search 
except ImportError:  
    print("Error/Not found") 
### Browsing the sourses using different search terms: 
search3_1 = "prevent contamination of uninfected patients"
search3_2 = "hospital spread prevention"
search3_3 = "prevent nosocomial outbreaks in hospitals"


##limit to the first 10 relevant links:
for a in search(search3_1, tld="co.in", num=10, stop=10, pause=2): 
    print(a)
for a in search(search3_2, tld="co.in", num=10, stop=10, pause=2): 
    print(a)
for a in search(search3_3, tld="co.in", num=10, stop=10, pause=2): 
    print(a)

https://www.cdc.gov/infectioncontrol/guidelines/isolation/prevention.html
https://www.ncbi.nlm.nih.gov/books/NBK2683/
https://www.who.int/water_sanitation_health/medicalwaste/148to158.pdf?ua=1
https://www.who.int/csr/resources/publications/whocdscsreph200212.pdf
https://www.health.ny.gov/professionals/diseases/reporting/communicable/infection/95-14_vre_control_guidelines.htm
https://www.cochrane.org/CD011621/protective-clothes-and-equipment-healthcare-workers-prevent-them-catching-coronavirus-and-other
http://www.bccdc.ca/resource-gallery/Documents/Guidelines%20and%20Forms/Guidelines%20and%20Manuals/Epid/CD%20Manual/Chapter%203%20-%20IC/InfectionControl_GF_IC_In_Physician_Office.pdf
https://www.halyardhealth.com/hai-watch/hai-threats-solutions/mrsa.aspx
http://www.southernhealth.nhs.uk/EasysiteWeb/getresource.axd?AssetID=83283&type=full&servicetype=Inline&filename=/Management_of_Used_and_Infected_Linen_V2.pdf
https://www.health.state.mn.us/facilities/patientsafety/infectioncontrol/pre/

In [50]:
#Manually choose the current relevant sources 
Q3_Online_Source={'url': ['https://www.who.int/water_sanitation_health/medicalwaste/148to158.pdf?ua=1',
                         'http://www.bccdc.ca/resource-gallery/Documents/Guidelines%20and%20Forms/Guidelines%20and%20Manuals/Epid/CD%20Manual/Chapter%203%20-%20IC/InfectionControl_GF_IC_In_Physician_Office.pdf',
                         'https://www.southernhealth.nhs.uk/EasysiteWeb/getresource.axd?AssetID=83283&type=full&servicetype=Inline&filename=/Management_of_Used_and_Infected_Linen_V2.pdf',
                         'https://www.cdc.gov/hai/prevent/tap.html',
                         'https://www.cdc.gov/hai/prevent/prevention_tools.html',
                         'https://www.cdc.gov/infectioncontrol/guidelines/isolation/prevention.html',
                         'https://www.cdc.gov/hai/prevent/prevention.html',
                         'https://www.cdc.gov/hai/prevent/infection-control-assessment-tools.html',
                         'https://www.ncbi.nlm.nih.gov/books/NBK2683/',
                          'https://www.health.ny.gov/professionals/diseases/reporting/communicable/infection/95-14_vre_control_guidelines.htm',
                          'https://www.cochrane.org/CD011621/protective-clothes-and-equipment-healthcare-workers-prevent-them-catching-coronavirus-and-other',
                          'https://www.health.state.mn.us/facilities/patientsafety/infectioncontrol/pre/contact.html',
                          'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3963198/'
                         ]}

In [54]:
Q3_Online_Source=pd.DataFrame.from_dict(Q3_Online_Source)
Q3_Online_Source.to_csv('E:/NorthEastern University/XN-Hackathon Covid/Kaggle Data/Q3_ZOnline_Sources.csv') 

# 4/ Methods to control the spread in communities, barriers to compliance

In [25]:
#Create different sets with specialized terms  from our questions 
Q4_a = tablesAbstract[tablesAbstract['abstract'].str.contains('compliance')]
Q4_b = tablesAbstract[tablesAbstract['abstract'].str.contains('community spread')]
Q4_c = tablesAbstract[tablesAbstract['abstract'].str.contains('prevent spread')]
Q4_d = tablesAbstract[tablesAbstract['abstract'].str.contains('methods to prevent')]

In [26]:
#Concatenitng the dataframes into one table per question and dropping null value 
Q4 = pd.concat([Q4_a,Q4_b,Q4_c,Q4_d])
Q4.dropna(inplace=True)
Q4.shape

(956, 6)

We have 956 articles regarding to the first topics at the moment

In [27]:
#Articles that we have
Q4.head(5)

title  \
145  A survey of knowledge, attitudes and practices...   
265  Heliox reduces respiratory system resistance i...   
520  Factors Affecting Intention to Receive and Sel...   
524  Autonomous Targeting of Infectious Superspread...   
892  PaCO(2 )and alveolar dead space are more relev...   

                                              abstract publish_time  \
145  BACKGROUND: Several public health strategic in...   2008-03-17   
265  INTRODUCTION: Respiratory syncytial virus (RSV...   2009-05-15   
520  BACKGROUND: Vaccination was a core component f...   2011-03-11   
524  Infectious disease treatments, both pharmaceut...   2011-03-17   
892  INTRODUCTION: Our aims in this study were to r...   2011-07-25   

                                                   url  \
145  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2...   
265  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2...   
520  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...   
524  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...   
892  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...   

                                               authors  \
145  Di Giuseppe, Gabriella; Abbate, Rossella; Alba...   
265  Kneyber, Martin CJ; van Heerde, Marc; Twisk, J...   
520  Liao, Qiuyan; Cowling, Benjamin J.; Lam, Wendy...   
524  Metzger, Vincent T.; Lloyd-Smith, James O.; We...   
892  Charron, Cyril; Repesse, Xavier; Bouferrache, ...   

                                                 words  
145  [BACKGROUND, Several, public, health, strategi...  
265  [INTRODUCTION, Respiratory, syncytial, virus, ...  
520  [BACKGROUND, Vaccination, was, a, core, compon...  
524  [Infectious, disease, treatments, both, pharma...  
892  [INTRODUCTION, Our, aims, in, this, study, wer...

In [28]:
## Saved as CSV file
Q4.to_csv('E:/NorthEastern University/XN-Hackathon Covid/Kaggle Data/Q4_Methods_control_spread_in_communities_and_barriers_to_compliance.csv')

Online Search

In [1]:
try: 
    from googlesearch import search 
except ImportError:  
    print("Error/Not found") 
### Browsing the sourses using different search terms: 
search4_1 = "prevent nosocomial outbreaks in hospitals"
search4_2 = "prevent COVID community spread"
search4_3 = "COVID 19 compliance"


##limit to the first 10 relevant links:
for a in search(search4_1, tld="co.in", num=10, stop=10, pause=2): 
    print(a)
for a in search(search4_2, tld="co.in", num=10, stop=10, pause=2): 
    print(a)
for a in search(search4_3, tld="co.in", num=10, stop=10, pause=2): 
    print(a)

https://nhcps.com/10-steps-preventing-spread-infection-hospitals/
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3963198/
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3963198/#idm140652065323648title
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3963198/#sec1-1title
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3963198/#sec1-2title
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3963198/#sec1-6title
https://www.cdc.gov/hai/prevent/prevention.html
https://www.cdc.gov/hai/prevent/infection-control-assessment-tools.html
https://www.cdc.gov/hai/prevent/prevention_tools.html
https://www.cdc.gov/hai/prevent/tap.html
https://www.cdc.gov/coronavirus/2019-ncov/prevent-getting-sick/prevention.html
https://www.cdc.gov/coronavirus/2019-ncov/prevent-getting-sick/cloth-face-cover-guidance.html
https://www.cdc.gov/coronavirus/2019-ncov/community/organizations/meat-poultry-processing-workers-employers.html
https://www.cdc.gov/niosh/topics/heatstress/default.html
https://wwwnc.cdc.gov/eid/article/26/7/20-1595

# 5/ What are recommendations for combating/overcoming resource failures

In [29]:
#Create different sets with specialized terms  from our questions 
Q5_a = tablesAbstract[tablesAbstract['abstract'].str.contains('not reach')]
Q5_b = tablesAbstract[tablesAbstract['abstract'].str.contains('improve access')]
Q5_c = tablesAbstract[tablesAbstract['abstract'].str.contains('access to resource')]
Q5_d = tablesAbstract[tablesAbstract['abstract'].str.contains('outreach')]
Q5_e = tablesAbstract[tablesAbstract['abstract'].str.contains('faulty')]
Q5_f = tablesAbstract[tablesAbstract['abstract'].str.contains('meet demand')]
Q5_g = tablesAbstract[tablesAbstract['abstract'].str.contains('waste')]

In [30]:
#Concatenitng the dataframes into one table per question and dropping null value 
Q5 = pd.concat([Q5_a,Q5_b,Q5_c,Q5_d, Q5_e,Q5_f,Q5_g])
Q5.dropna(inplace=True)
Q5.shape

(813, 6)

We have 813 articles regarding to the first topics at the moment

In [31]:
#Articles that we have
Q5.head(5)

title  \
1480  Building Cell Selectivity into CPP-Mediated St...   
2014  Is There Any Role of Inhalational Corticostero...   
2067  Quantity and Size Distribution of Cough-Genera...   
2359  Common Genetic Polymorphisms within NFκB-Relat...   
2904  Effectiveness of a fluid chart in outpatient m...   

                                               abstract publish_time  \
1480  There is a pressing need for more effective an...   2010-05-14   
2014  Fat embolism syndrome (FES) is primarily a lun...   2015-09-25   
2067  The question of whether influenza is transmitt...   2012-05-31   
2359  Invasive Aspergillosis (IA) is an opportunisti...   2016-08-12   
2904  INTRODUCTION: Dengue infection is the fastest ...   2017-10-04   

                                                    url  \
1480  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4...   
2014  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4...   
2067  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4...   
2359  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4...   
2904  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...   

                                                authors  \
1480  Martín, Irene; Teixidó, Meritxell; Giralt, Ernest   
2014  Agarwal, Amit Kumar; Sen, Ramesh; Tripathy, Su...   
2067  Lindsley, William G.; Pearce, Terri A.; Hudnal...   
2359  Lupiañez, Carmen B.; Villaescusa, María T.; Ca...   
2904  Nasir, Nazrila Hairin; Mohamad, Mohazmi; Lum, ...   

                                                  words  
1480  [There, is, a, pressing, need, for, more, effe...  
2014  [Fat, embolism, syndrome, FES, is, primarily, ...  
2067  [The, question, of, whether, influenza, is, tr...  
2359  [Invasive, Aspergillosis, IA, is, an, opportun...  
2904  [INTRODUCTION, Dengue, infection, is, the, fas...

In [32]:
## Saved as CSV file
Q5.to_csv('E:/NorthEastern University/XN-Hackathon Covid/Kaggle Data/Q5_Recommendations_for_combating_resource_failures.csv')

Online Search

In [3]:
try: 
    from googlesearch import search 
except ImportError:  
    print("Error/Not found") 
### Browsing the sourses using different search terms: 
search5_1 = "recommendations for COVID 19 resources limits"
search5_2 = "recommendations for COVID 19 testing problems"
search5_3 = "recommendations for PPE problems"
search5_4 = "recommendations for improving access to COVID 19 resources"



##limit to the first 10 relevant links:
for a in search(search5_1, tld="co.in", num=10, stop=10, pause=2): 
    print(a)
for a in search(search5_2, tld="co.in", num=10, stop=10, pause=2): 
    print(a)
for a in search(search5_3, tld="co.in", num=10, stop=10, pause=2): 
    print(a)
for a in search(search5_4, tld="co.in", num=10, stop=10, pause=2): 
    print(a)

https://www.cdc.gov/coronavirus/2019-ncov/community/large-events/index.html
https://www.cdc.gov/coronavirus/2019-ncov/community/large-events/event-planners-and-attendees-faq.html
https://www.cdc.gov/coronavirus/2019-ncov/community/large-events/considerations-for-events-gatherings.html
https://www.kingcounty.gov/depts/health/covid-19.aspx
https://www.kingcounty.gov/depts/health/covid-19/data.aspx
https://kingcounty.gov/depts/health/covid-19/care/testing.aspx
https://www.kingcounty.gov/depts/health/covid-19/care.aspx
https://www.doh.wa.gov/coronavirus/workplace
https://www.idsociety.org/contentassets/9ba35522e0964d51a47ae3b22e59fb47/idsa-recommendations-for-reducing-covid-19-distancing_16apr2020_final-.pdf
https://www.wsha.org/for-patients/coronavirus/coronavirus-resources-for-hospitals/
https://www.cdc.gov/coronavirus/2019-ncov/lab/testing.html
https://www.fda.gov/medical-devices/letters-health-care-providers/certain-covid-19-serologyantibody-tests-should-not-be-used-letter-clinical-lab